genesyscloud/tfexporter/genesyscloud_resource_exporter.go

```diff
@@ -429,6 +422,65 @@ func (g *GenesysCloudResourceExporter) buildResourceConfigMap() diag.Diagnostics
 	return nil
 }
 
+func (g *GenesysCloudResourceExporter) customWriteAttributes(jsonResult util.JsonMap,
+	resource resourceExporter.ResourceInfo) {
+	exporters := *g.exporters
+	if resourceFilesWriterFunc := exporters[resource.Type].CustomFileWriter.RetrieveAndWriteFilesFunc; resourceFilesWriterFunc != nil {
+		exportDir, _ := getFilePath(g.d, "")
+		if err := resourceFilesWriterFunc(resource.State.ID, exportDir, exporters[resource.Type].CustomFileWriter.SubDirectory, jsonResult, g.meta); err != nil {
+			log.Printf("An error has occurred while trying invoking the RetrieveAndWriteFilesFunc for resource type %s: %v", resource.Type, err)
+		}
+		g.updateInstanceStateAttributes(jsonResult, resource)
+	}
+
+	if len(exporters[resource.Type].CustomFlowResolver) > 0 {
+		g.updateInstanceStateAttributes(jsonResult, resource)
+	}
+}
+
+func (g *GenesysCloudResourceExporter) updateInstanceStateAttributes(jsonResult util.JsonMap,
+	resource resourceExporter.ResourceInfo) {
+
+	for attr, val := range jsonResult {
```

Comment:
This seems like a really big convuluted if..for..if..if..for..if loop. We might want to reconsider refactoring this or at the least put a ticket out there to refactor this.  :)

Reply:
refactored this code logic a bit.